### PR TITLE
Support XGBClassifier with pandas DataFrame in PermutationImportance

### PR DESCRIPTION
**Pull request by hofesh**
fix for issue:
https://github.com/TeamHG-Memex/eli5/issues/256

### CODE REVIEW

File: eli5/sklearn/permutation_importance.py [hunk 1]
```diff
@@ -15,7 +15,10 @@
 from sklearn.metrics.scorer import check_scoring  # type: ignore
 
 from eli5.permutation_importance import get_score_importances
+from eli5.sklearn.utils import pandas_available
 
+if pandas_available:
+    import pandas as pd
```
Comment (lopuhin) [inline eli5/sklearn/permutation_importance.py]: Looks like the build failure https://travis-ci.org/TeamHG-Memex/eli5/jobs/492103801#L781 is due to missing   ``# type: ignore`` commend at the end here:
```suggestion
    import pandas as pd   # type: ignore
```

**Comment by hofesh** [comment]
@lopuhin added a unit test. not sure about your assumption about having pandas and xgboost when running the tests.

File: tests/test_sklearn_permutation_importance.py [hunk 2]
```diff
@@ -9,11 +9,13 @@
 from sklearn.feature_selection import SelectFromModel
 from sklearn.linear_model import LogisticRegression
 
+from xgboost import XGBClassifier # type: ignore
+import pandas as pd # type: ignore
```
Comment (lopuhin) [inline tests/test_sklearn_permutation_importance.py]: xgboost and pandas are indeed available not in all test environments, to fix this it's possible to put the imports into the test function like this (see https://docs.pytest.org/en/latest/skipping.html#skipping-on-a-missing-import-dependency):

```
xgboost = pytest.importorskip('xgboost')
pd =  pytest.importorskip('pandas')
```

(also ``# type: ignore`` is not required here, as tests are not type-checked at the moment)

Comment (hofesh) [reply tests/test_sklearn_permutation_importance.py]: fixed

File: eli5/sklearn/permutation_importance.py [hunk 3]
```diff
@@ -151,6 +154,12 @@ def __init__(self, estimator, scoring=None, n_iter=5, random_state=None,
         self.cv = cv
         self.rng_ = check_random_state(random_state)
 
+    def wrap_scorer(self, base_scorer, pd_columns):
```
Comment (kmike) [inline eli5/sklearn/permutation_importance.py]: would you mjind making this method private (`_wrap_scorer`)? I'm not sure it should be a part of the public API

File: eli5/sklearn/permutation_importance.py [hunk 4]
```diff
@@ -182,6 +191,9 @@ def fit(self, X, y, groups=None, **fit_params):
             self.estimator_ = clone(self.estimator)
             self.estimator_.fit(X, y, **fit_params)
 
+        if pandas_available and isinstance(X, pd.DataFrame):
+            self.scorer_ = self.wrap_scorer(self.scorer_, X.columns)
```
Comment (kmike) [inline eli5/sklearn/permutation_importance.py]: I think it would be better to move all `self.scorer_` - related code close to each other - maybe move `self.scorer_ = check_scoring(self.estimator, scoring=self.scoring)` line just before this new code, so that they're not separated by `prefit` handling code?

File: tests/test_sklearn_permutation_importance.py [hunk 5]
```diff
@@ -13,7 +13,6 @@
 from eli5.sklearn import PermutationImportance
 from .utils import format_as_all
 
-
```
Comment (kmike) [inline tests/test_sklearn_permutation_importance.py]: according to pep8, function should be separated by two lines, not but one - could you please keep the line here, and add a line before the new test function?

**Comment by kmike** [comment]
I left a few minor comments, but the code looks good to me - thanks @hofesh!

**Comment by hofesh** [comment]
@kmike fixed your comments

**Comment by kmike** [comment]
Thanks @hofesh for the fix!
